### ISSUE TITLE
Bug: [no-unused-vars] False positive for types referenced in jsDoc

### ISSUE THREAD

**Issue by matthieusieben** [typescript-eslint/typescript-eslint#12101]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=JYWwDg9gTgLgBAbzgMQhOBfOAzKERwDkAdAPTZqEBQVpAVHVXHXAAIDOApp4qwDbAAdgGsUaDEzqkqnAB6RYOAK6CAxjGARBONAAoAlIglA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
import { Foo } from './foo'

/**
 * @see {@link Foo}
 */
export function foo() {}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expect the `Foo` type *not* to be reported as unused by eslint.

### Actual Result

The `Foo` type *is* reported as unused by eslint.

### Additional Info

It is actually used by typescript itself. In particular IDE's will render an href for the jsDoc `{@link x}` values, which will open the referenced type when clicked.

It would be nice to at least have an option to allow types to be considered "used" when referenced from jsDoc.

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/11208, #10873, #9435, #8902, #5017, #6967, #7944, #8258

Please use [the search](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue%20state%3Aclosed%20%22no-unused-vars%22%20jsdoc) before filing an issue.

**Comment by matthieusieben** [comment]
Maybe you should consider adding this to the FAQ if this is a frequently asked feature that you don't want to support.
